oneclick.py

```diff
@@ -0,0 +1,600 @@
+        authorization = base64.b64encode(authorization)
+        return dict(Authorization="Basic " + authorization)
+
+
+    def _make_request(self, url, method, headers, data=None, params=None, **kwargs):
+        print "url= %s" % url
+        print "params= %s" % params
+        print "kwargs= %s" % kwargs
+
+        """Actually make an HTTP request."""
+        return HTTP.request_with_timeout(
+            method, url, headers=headers, data=data,
+            params=params, **kwargs
+        )
+
+
+    def request(self, url, method='get', extra_headers={}, data=None,
+                params=None, verbosity='complete'):
+        """Make an HTTP request, acquiring/refreshing a bearer token
+        if necessary.
+        """
+        if verbosity not in self.RESPONSE_VERBOSITY.values():
+            verbosity = self.RESPONSE_VERBOSITY[2]
+
+        headers = dict(extra_headers)
+        headers['Authorization'] = "Basic " + self.token
+        headers['Content-Type'] = 'application/json'
+        headers['Accept-Media'] = verbosity
+
+        # for now, do nothing with error codes, but in the future might have some that 
+        # will warrant repeating the request.
+        disallowed_response_codes = ["409"]
+        response = self._make_request(
+            url=url, method=method, headers=headers,
+            data=data, params=params, 
+            disallowed_response_codes=disallowed_response_codes
+        )
+        
+        return response
+
+
+    ''' --------------------- Getters and Setters -------------------------- '''
+
+    '''
+    TODO:
+    library delta:
+    http://api.oneclickdigital.us/v1/libraries/1998/media/delta?begin=2016-07-26&end=2016-09-26
+    get
+
+    get whole catalog
+    https://api.oneclickdigital.us/v1/libraries/1931/media/all
+    get
+    '''
+
+    def get_all_available_through_search(self):
+        """
+        Gets a list of ebook and eaudio items this library has access to, that are currently
+        available to lend.  Uses the "availability" facet of the search function.
+        An alternative to self.get_availability_info().
+        Calls paged search until done.
+        Uses minimal verbosity for result set.
+
+        Note:  Some libraries can see other libraries' catalogs, even if the patron 
+        cannot checkout the items.  The library ownership information is in the "interest" 
+        fields of the response.
+
+        :return A dictionary representation of the response, containing catalog count and ebook item - interest pairs.
+        """
+        page = 0;
+        response = self.search(availability='available', verbosity=self.RESPONSE_VERBOSITY[0])
+
+        respdict = response.json()
+        if not respdict:
+            raise IOError("OneClick availability response not parseable - has no respdict.")
+
+        if not ('pageIndex' in respdict and 'pageCount' in respdict):
+            raise IOError("OneClick availability response not parseable - has no page counts.")
```

Comment:
Looked into it, the RemoteInitiatedServerError is defined in circulation_exceptions.  I'll keep it IOError in core, and then call the better error from the circulation oneclick code.